core/settings_dialog.py

```
@@ -9,125 +9,30 @@
-                self.error.emit(f"Download failed:\n{e}")
-                return
-
-        self.error.emit(
-            f"Could not download {owner}/{repo} (tried main and master branches).\n{last_error}"
-        )
-
-
 class SettingsDialog(QDialog):
     """Settings dialog"""
 
     def __init__(self, settings: Dict[str, Any], parent=None, available_modules: List[tuple] = None,
```

Ruff (0.15.9)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)